src/libraries/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -488,6 +489,39 @@ public static IEnumerable<string> ReadLines(string path, Encoding encoding)
             return ReadLinesIterator.CreateIterator(path, encoding);
         }
 
+        public static IAsyncEnumerable<string> ReadLinesAsync(string path, CancellationToken cancellationToken = default)
+        {
+            return ReadLinesAsync(path, Encoding.UTF8, cancellationToken);
+        }
+
+        public static IAsyncEnumerable<string> ReadLinesAsync(string path, Encoding encoding, CancellationToken cancellationToken = default)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            return InternalReadLinesAsync(path, encoding, cancellationToken);
+        }
+
+        private static async IAsyncEnumerable<string> InternalReadLinesAsync(string path, Encoding encoding, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+            Debug.Assert(encoding != null);
+
+            using StreamReader sr = AsyncStreamReader(path, encoding);
```

Comment:
File.ReadLines has some special behaviors related to this.  It actually creates the StreamReader as part of the ReadLines call, so that the file is opened as part of the ReadLines call and any validation as part of that (e.g. does the file exist) happens as part of the call to ReadLines, rather than as part of the first MoveNext on the resulting iterator.  It then has to take care to create a new reader on subsequent iterations.  For consistency between ReadLines and ReadLinesAsync, and for all the reasons ReadLines chose that approach, we should probably do the same thing here, even though it means extra complication.